File: src/models/user.model.ts
import { Schema, model } from 'mongoose';
import { IUserInterface } from '../interfaces/user.interface';

const user = new Schema<IUserInterface>({
       phoneNumber: {
        type: String,
        unique: true, // Assuming email should be unique
        validate: {
          validator: function(v: string) {
            // Example: Validate if the phone number follows a specific format
            return /\d{3}-\d{3}-\d{4}/.test(v);
          },
          message: props => `${props.value} is not a valid phone number!`
        },
    },
    password: {
        type: String,
        minlength: [8, 'Password should have at least 8 characters'],
        required: [true, 'Password is required'],
        select: false,
    }, 
    role: { type: String, enum: ['Admin', 'User'], required: true },
    verified: { type: Boolean, required: true },
    chapaAccountId: { type: String },
},
{
    timestamps: true,
});

export default model<IUserInterface>('User', user);

File: frontend/lib/models/user_model.dart
class User{
  final String firstName;
  final String lastName;
  final String phoneNumber;
  final String city;
  final String password;
  final String confirmPassword;

  User({
    required this.firstName,
    required this.lastName,
    required this.phoneNumber,
    required this.city,
    required this.password,
    required this.confirmPassword,
  });

  Map<String, dynamic> toJson() {
    return {
      "firstName": firstName,
      "lastName": lastName,
      "phoneNumber": phoneNumber,
      "city": city,
      "password": password,
      "confirmPassword": confirmPassword,
    };
  }
}
